src/ChainRulesTestUtils.jl

```diff
@@ -12,11 +12,21 @@ using Test
 import FiniteDifferences: rand_tangent
 
 const _fdm = central_fdm(5, 1; max_range=1e-2)
+const TEST_INFERRED = Ref(true)
 
 export TestIterator
 export test_approx, test_scalar, test_frule, test_rrule, generate_well_conditioned_matrix
 export ⊢
+export @maybe_inferred
 
+function __init__()
+    TEST_INFERRED[] = !parse(Bool, get(ENV, "JULIA_PKGEVAL", "false")) &&
+        parse(Bool, get(ENV, "CHAINRULES_TEST_INFERRED", "true"))
```

Comment:
Possible we want:
```
TEST_INFERRED[] = if haskey(ENV, "CHAINRULES_TEST_INFERRED")
    parse(Bool, "CHAINRULES_TEST_INFERRED")
else
    !parse(Bool, get(ENV, "JULIA_PKGEVAL", "false"))
end
```

which as truth table
for C being `ENV[CHAINRULES_TEST_INFERRED]` and
`P` being `JULIA_PKGEVAL`
and `X` marking not defined

```
C P R
-----
X T F
X F T
F F F
F T F
T F T
T T T
T X T
F X F
X X T
```

over 
```
TEST_INFERRED[] = !parse(Bool, get(ENV, "JULIA_PKGEVAL", "false")) &&
    parse(Bool, get(ENV, "CHAINRULES_TEST_INFERRED", "true"))
```
which has
```
C P R
-----
X T F
X F T
F F F
F T F
T F T
T T F
T X T
F X F
X X T
```

The difference is the 
`T T T` vs `T T F`.
I.e. if the ChainRules specific environment variable is present, it would override the `JULIA_PKGEVAL`.
Which feels right to me.
If someone specifically states they want these tests to run that is stronger than if it happens to have this enviroment varaible set.

Also do we want to handle the variables not being set to parsable values?
If so could be done with `try_parse` + `something`, or the a try-catch + give an informative message.

Reply:
Maybe we don't need the two env-vars.
Maybe we just need `!parse(Bool, get(ENV, "JULIA_PKGEVAL", "false"))` to see the default value.
And then for normal users (and for overwriting) we can document doing `ChainRulesTestUtils.TEST_INFERRED[]=false` / `ChainRulesTestUtils.TEST_INFERRED[]=true`

Maybe that is better?

Reply:
I think a separate environment variable might still be useful, since otherwise, you'd have to manually edit the testing code to disable these. I didn't really expect the two env vars to ever really be set at the same time, but just for consistency's sake I agree that your version makes more sense, so I will go with that instead if that's ok.